src/Core/Services/Implementations/CipherService.cs

```diff
@@ -597,22 +569,8 @@ public async Task DeleteFolderAsync(Folder folder)
             var cipherIds = new List<Guid>();
             foreach (var (cipher, lastKnownRevisionDate) in cipherInfos)
             {
-                if (cipher.Id == default(Guid))
-                {
-                    throw new BadRequestException("All ciphers must already exist.");
-                }
-
-                if (cipher.OrganizationId.HasValue)
-                {
-                    throw new BadRequestException("One or more ciphers already belong to an organization.");
-                }
-
-                if (!cipher.UserId.HasValue || cipher.UserId.Value != sharingUserId)
-                {
-                    throw new BadRequestException("One or more ciphers do not belong to you.");
-                }
-
-                ValidateCipherLastKnownRevisionDateAsync(cipher, lastKnownRevisionDate);
+                var attachments = cipher.GetAttachments();
```

Comment:
I left it out of `ValidateCipherCanBeShared` because it is used again in the `ShareAsync()` method, so if I move it into `ValidateCipherCanBeShared`, we'll still have to call it again in ShareAsync (which might not be a big deal)

Reply:
Not a big deal imo - it's worth it to reduce the number of arguments in `ValidateCipherCanBeShared`, unless there is any specific performance concern (but it's just deserializing a small amount of data on the Cipher object so I think that's fine). I think it's generally an anti-pattern to pass in an argument that can be derived from another, already existing argument.

Reply:
I went ahead and moved it, if this was in the ShareMany method I think it would have been better to leave it, but it's on a single cipher so it won't have much of any impact at all if it makes the code cleaner :)